server/src/main/java/org/elasticsearch/action/admin/cluster/state/TransportClusterStateAction.java

```diff
@@ -115,6 +116,13 @@ public void onTimeout(TimeValue timeout) {
         }
     }
 
+    // The cluster state can be a large object so we resolve the response listener on the MANAGEMENT pool to not serialize the resulting
+    // large response on a transport or cluster state thread which we don't want to block needlessly
+    private void resolveListener(ClusterStateRequest request, ClusterState currentState, ActionListener<ClusterStateResponse> listener) {
+        threadPool.executor(ThreadPool.Names.MANAGEMENT).execute(
```

Comment:
If we build a response not on a transport thread, we still serialise it on the transport thread right? If it's just the outbound serialization bit that's slow, how is it logged? AIUI today we only log slowness when processing an inbound request.

Reply:
> If we build a response not on a transport thread, we still serialise it on the transport thread right? 

Nah not anymore, we reverted that change. Once again serialization happens on the thread that resolves the listener in the transport action.

Reply:
Oh yeah, sorry, I lost track. This is ok then, at least until we reinstate that change.